src/librustc_resolve/lib.rs

```diff
@@ -1170,7 +1167,7 @@ pub struct Resolver<'a, 'tcx: 'a> {
 
     ast_map: &'a hir_map::Map<'tcx>,
```

Comment:
This seems to infect everything with `'tcx` but its improper, it should be `'hir` instead (`'tcx` is for the type context, which we do not have here).